experiments/src/matcher.rs

```diff
@@ -0,0 +1,36 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+//! This module defines all the information needed to match a user with an experiment
+//! this module should also include a `match` function of some sort that does the matching
+//! it has two main types, the `matcher` retrieved from the server, and the `AppContext`
+//! from the client
+//! Note: This could be where the logic to evaluate the filter_expressions lies
+use serde_derive::*;
+
+#[derive(Deserialize, Serialize, Debug, Clone, Default)]
+pub struct Matcher {
+    pub app_id: Option<String>,
```

Comment:
@pdehaan do you have a list of fields that Firefox Desktop implementation provides for Experiment context? Are we missing anything?

Reply:
I believe Desktop is using JEXL for their targeting/matching, and the only docs I found on that [are here](https://firefox-source-docs.mozilla.org/browser/components/newtab/content-src/asrouter/docs/targeting-guide.html?highlight=jexl%20targeting)